modules/tinymce/src/plugins/table/test/ts/browser/EmptyRowTableTest.ts

```diff
@@ -0,0 +1,41 @@
+import { Log, Pipeline, Step } from '@ephox/agar';
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import { TinyApis, TinyLoader } from '@ephox/mcagar';
+import { Selectors, SugarElement } from '@ephox/sugar';
+
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+UnitTest.asynctest('browser.tinymce.plugins.table.EmptyRowTableTest', (success, failure) => {
+  Theme();
+  Plugin();
+
+  TinyLoader.setupLight((editor, success, failure) => {
+    const tinyApis = TinyApis(editor);
+
+    Pipeline.async({}, [
+      Log.stepsAsStep('TINY-4679', 'Empty tr elements should not be removed', [
+        tinyApis.sSetContent(`
+          <div>
+            <table>
+              <colgroup><col></colgroup>
+              <tbody>
+                <tr><td rowspan="2" >TR 1</td></tr>
+                <tr></tr>
```

Comment:
I wonder if it's worth checking any weird things like a `br` inside the `tr`? I'm not actually sure what the result would be for that.

Reply:
In my testing, anything inside the `tr` would result in it _not_ being removed (including commented HTML). As far as I can tell, anything inside the `tr` now will not change the structure of the markup, but could still have layout consequences outside the scope of this ticket

Reply:
From MDN, the only things allowed inside a tr are:
> Zero or more `<td>` and/or `<th>` elements; script-supporting elements (`<script>` and `<template>`) are also allowed